scripts/irods/test/test_pam_password_authentication.py

```diff
@@ -82,83 +89,102 @@ def setUpClass(self):
+        self.other_auth_session.__exit__()
+        self.admin.__exit__()
+
+        # Put all the modified configurations back and reload the server configuration.
+        shutil.copyfile(self.service_account_environment_file_backup, self.service_account_environment_file_path)
+        shutil.copyfile(self.server_config_backup, paths.server_config_path())
+        shutil.copyfile(self.core_re_file_backup, self.core_re_path)
+        IrodsController().reload_configuration()
 
         for filename in [self.chain_pem_path, self.server_key_path, self.dhparams_pem_path]:
             if os.path.exists(filename):
                 os.unlink(filename)
 
-        self.admin.assert_icommand(['iadmin', 'rmuser', self.auth_session.username])
-        self.admin.__exit__()
         with session.make_session_for_existing_admin() as admin_session:
-            admin_session.assert_icommand(['iadmin', 'rmuser', self.admin.username])
+            admin_session.run_icommand(['iadmin', 'rmuser', self.auth_session.username])
+            admin_session.run_icommand(['iadmin', 'rmuser', self.admin.username])
 
     def do_test_invalid_password_time_configurations(self, _option_name):
         # Stash away the original configuration for later...
         original_config = self.admin.assert_icommand(
                 ['iadmin', 'get_grid_configuration', self.configuration_namespace, _option_name], 'STDOUT')[1].strip()
 
-        IrodsController().stop()
-
         auth_session_env_backup = copy.deepcopy(self.auth_session.environment_file_contents)
-        admin_session_env_backup = copy.deepcopy(self.admin.environment_file_contents)
         try:
-            service_account_environment_file_path = os.path.join(
-                os.path.expanduser('~'), '.irods', 'irods_environment.json')
-            with lib.file_backed_up(service_account_environment_file_path):
-                client_update = test_configurations.make_dict_for_ssl_client_environment(
-                    self.server_key_path, self.chain_pem_path, self.dhparams_pem_path)
-
-                lib.update_json_file_from_dict(service_account_environment_file_path, client_update)
-
-                self.admin.environment_file_contents.update(client_update)
+            client_update = {'irods_authentication_scheme': self.authentication_scheme}
+            self.auth_session.environment_file_contents.update(client_update)
 
-                client_update['irods_authentication_scheme'] = self.authentication_scheme
-                self.auth_session.environment_file_contents.update(client_update)
-
-                with temporary_core_file() as core:
-                    core.add_rule(test_configurations.get_pep_for_ssl(self.plugin_name))
+            # Make sure the settings applied correctly...
+            self.admin.assert_icommand(
+                ['iadmin', 'get_grid_configuration', self.configuration_namespace, _option_name],
+                'STDOUT', original_config)
```

Comment:
This assertion seems a bit strange. I don't see where the value is being changed? I haven't spotted how this can fail.

_I noticed this in other tests too._

Reply:
Note: I did not change the assertions in any of these tests. All that changed is how the TLS is being configured for the test class. It's easier to see this if you hide whitespace changes.

Anyway, this assertion seems to be a sanity check to ensure that the TLS configuration which was previously occurring was working for both the admin and test session. It probably isn't needed.

None of the actual assertions in the tests touched in this PR were modified except where necessary as it related directly to the changes being made to TLS configuration.

Reply:
Makes sense.

Consider adding a commit which removes those assertions. Running only the tests which are affected is enough for that. No rerun of the full test suite necessary.

Reply:
Done. Test passes